src/compiler/emitter.ts

```diff
@@ -4707,6 +4707,9 @@ if (typeof __param !== "function") __param = function (paramIndex, decorator) {
                 }
 
                 write("[\"require\", \"exports\"");
+                if (compilerOptions.noEmitExtends) {

```

Comment:
This could then be defined and included by the user however they see fit. The TypeScript document could be updated to include this *boilerplate* if this flag gets the greenlit.